tools/bin/check_images_exist.sh

```diff
@@ -1,78 +1,137 @@
 #!/usr/bin/env bash
 
-set -e
+# ------------- Import some defaults for the shell
+
+# Source shell defaults
+# $0 is the currently running program (this file)
+this_file_directory=$(dirname $0)
+relative_path_to_defaults=$this_file_directory/../shell_defaults
+
+# if a file exists there, source it. otherwise complain
+if test -f $relative_path_to_defaults; then
+  # source and '.' are the same program
+  source $relative_path_to_defaults
+else
+  echo -e "\033[31m\nFAILED TO SOURCE TEST RUNNING OPTIONS.\033[39m"
+  echo -e "\033[31mTried $relative_path_to_defaults\033[39m"
+  exit 1
+fi
+
+set +o xtrace  # +x easier human reading here
 
 . tools/lib/lib.sh
 
+
 function docker_tag_exists() {
-  # Added check for images pushed to github container registry
-  if [[ $1 == ghcr* ]]
+  # Is true for images stored in the Github Container Registry
+  repo=$1
+  tag=$2
+  # we user [[ here because test doesn't support globbing well
+  if [[ $repo == ghcr* ]]
   then
     TOKEN_URL=https://ghcr.io/token\?scope\="repository:$1:pull"
-    token=$(curl $TOKEN_URL | jq -r '.token')
+    token=$(curl $TOKEN_URL | jq -r '.token' > /dev/null)
     URL=https://ghcr.io/v2/$1/manifests/$2
-    printf "\tURL: %s\n" "$URL"
-    curl --silent -H "Authorization: Bearer $token" -f -lSL "$URL" > /dev/null
+    echo -e "$blue_text""\n\n\tURL: $URL""$default_text"
+    curl -H "Authorization: Bearer $token" --location --silent --show-error --dump-header header.txt "$URL" > /dev/null
+    curl_success=$?
   else
     URL=https://hub.docker.com/v2/repositories/"$1"/tags/"$2"
-    printf "\tURL: %s\n" "$URL"
-    curl --silent -f -lSL "$URL" > /dev/null
+    echo -e "$blue_text""\n\n\tURL: $URL""$default_text"
+    curl --silent --show-error --location --dump-header header.txt "$URL" > /dev/null
+    curl_success=$?
+    # some bullshit to get the number out of a header that looks like this
+    # < content-length: 1039
+    # < x-ratelimit-limit: 180
+    # < x-ratelimit-reset: 1665683196
+    # < x-ratelimit-remaining: 180
+    docker_rate_limit_remaining=$(grep 'x-ratelimit-remaining: ' header.txt | grep --only-matching --extended-regexp "\d+")
+    # too noisy when set to < 1.  Dockerhub starts complaining somewhere around 10
+    if test "$docker_rate_limit_remaining" -lt 20; then
+      echo -e "$red_text""We are close to a sensitive dockerhub rate limit!""$default_text"
+      echo -e "$red_text""SLEEPING 60s sad times""$default_text"
+      sleep 60
+      docker_tag_exists $1 $2
+    elif test $docker_rate_limit_remaining -lt 50; then
```

Comment:
I think this might be doing a weird thing? but not in a way that crashes the script https://github.com/airbytehq/airbyte/actions/runs/3330150392/jobs/5508261510

```
./tools/bin/check_images_exist.sh: line 50: test: : integer expression expected
	STATUS: found
	airbyte/destination-redshift: 0.3.50

./tools/bin/check_images_exist.sh: line 55: test: -lt: unary operator expected
```

not sure if this is causing false positives, but at least the check still correctly passes when the images _do_ exist